File: problemas/iniciante/1065/1065.cpp
#include <iostream>

using namespace std;

int main()
{
    int pares, numero;

    pares = 0;
    for (int i = 0; i < 5; ++i)
    {
        cin >> numero;

        if (numero % 2 == 0)
        {
            pares += 1;
        }
    }

    cout << pares << " valores pares" << endl;

    return 0;
}

File: problemas/iniciante/1065/1065.hs
import Text.Printf

pares :: Int -> IO Int -> IO Int
pares 0 result = result
pares x result = do
  numero <- readLn :: IO Int
  if mod numero 2 == 0
    then pares (x - 1) (result >>= \r -> return (r + 1))
    else pares (x - 1) result

main :: IO ()
main = do
  resposta <- pares 5 (return 0)
  printf "%d valores pares\n" resposta
